amethyst_rendy/src/render_test_bundle.rs

```diff
@@ -0,0 +1,181 @@
+use std::{ops::Deref, sync::Arc};
+
+use amethyst_assets::Processor;
+use amethyst_core::{
+    bundle::SystemBundle,
+    ecs::{DispatcherBuilder, ReadExpect, Resources, SystemData},
+};
+use amethyst_error::Error;
+use amethyst_window::{ScreenDimensions, Window};
+use derive_new::new;
+
+use crate::{
+    pass::{DrawFlat2DDesc, DrawFlat2DTransparentDesc},
+    rendy::{
+        factory::Factory,
+        graph::{
+            present::PresentNode,
+            render::{RenderGroupDesc, SubpassBuilder},
+            GraphBuilder,
+        },
+        hal::{
+            command::{ClearDepthStencil, ClearValue},
+            format::Format,
+            image::Kind,
+        },
+    },
+    sprite::SpriteSheet,
+    sprite_visibility::SpriteVisibilitySortingSystem,
+    types::DefaultBackend,
+    GraphCreator, RenderingSystem,
+};
+
+/// Adds sprite systems and a basic rendering system to the dispatcher.
+///
+/// This test bundle requires the user to also add the `TransformBundle` and `WindowBundle` to the
+/// dispatcher.
+#[derive(Debug, new)]
+pub struct RenderTestBundle;
+
+impl<'a, 'b> SystemBundle<'a, 'b> for RenderTestBundle {
+    fn build(self, builder: &mut DispatcherBuilder<'a, 'b>) -> Result<(), Error> {
+        builder.add(
+            Processor::<SpriteSheet>::new(),
+            "sprite_sheet_processor",
+            &[],
+        );
+        builder.add(
+            SpriteVisibilitySortingSystem::new(),
+            "sprite_visibility_system",
+            &["transform_system"],
+        );
+
+        builder.add_thread_local(RenderingSystem::<DefaultBackend, _>::new(
```

Comment:
`DefaultBackend` isn't really safe to be used here. This type alias is unavailable when you compile with multiple backends. We don't want to limit testing to one backend only.

Reply:
I've replaced it with a type parameter, is that okay?
(used the "i don't know what i'm doing, but it compiles" strategy)